app/models/manageiq/providers/ibm_cloud/inventory/collector/object_storage.rb

```diff
@@ -0,0 +1,29 @@
+class ManageIQ::Providers::IbmCloud::Inventory::Collector::ObjectStorage < ManageIQ::Providers::IbmCloud::Inventory::Collector
+  require_nested :StorageManager
+
+  def buckets
+    buckets = []
+
+    # XXX: filtering out buckets not from our region, as of now the REST call fails with an
+    # exception if used with from different region than the bucket's
+    (connection.list_buckets[:buckets] || []).each do |bucket|
+      connection.get_bucket_location(:bucket => bucket[:name]).location_constraint
+      buckets << bucket
+    rescue
+      _log.warn("bucket '#{bucket[:name]}' is not from our region '#{manager.provider_region}', skipping")
```

Comment:
Is a bucket not in our region something that the user has misconfigured or is it totally normal?  If it is normal than warning is likely cause more alarm than necessary.  If normal we might not even need a log at all.

Reply:
Seems like "list_bucket" returns buckets from all regions in a given instance and not from the current one. We need to filter those out as subsequent calls (e.g. list_objects, etc) will fail if used with buckets from a foreign region. Seems to be some arch. issue on the S3 side, I'm not totally sure. Similar issue is described here https://github.com/aws/aws-sdk-go/issues/720.